### ISSUE TITLE
Allow for more sorting options within Albums view

### ISSUE THREAD

**Issue by XTREEMMAK** [pulsejet/memories#558]
**Is your feature request related to a problem? Please describe.**
At present, within Albums view, you are currently limited to two sorting options; "By Date" and "By Name". The unfortunate side is that you can't specify in which direction those sorts take place aka Ascending or Descending (A -> Z or Z -> A). Also, soring by date is based on the Album's creating date, regardless of when those images were shot.

**Describe the solution you'd like**
I'd like to see more sorting options (Ascending, Descending, perhaps Highest Rated, By Number of Photos, etc), and possibly a custom key/value for albums so you can specify a custom date/time parameter that the sort options will use instead aka, letting the user specify the date specifics manually.

**Describe alternatives you've considered**
You can use the Browser's search option to find the album text that you're looking for, but this is definitely not very intuitive.

**Comment by Rollbacke** [comment]
Hi team,
I'm taking this thread to also add the ability to save preferred sort order.
When we change the sort order, at the next connection, it always returns to "sort by date".

**Comment by pulsejet** [comment]
@Rollbacke this is a local setting, it'll only persist in your browser.

**Comment by Rollbacke** [comment]
Hi @pulsejet, precisely. What I propose is to add it to the settings so that the option persists outside the browser. :)

**Comment by ldaverio** [comment]
> Hi @pulsejet, precisely. What I propose is to add it to the settings so that the option persists outside the browser. :)

I second that suggestion :) A reverse chronological order may make sense in a timeline, but not so much in an album, when you want people to explore your photos in the same order as they were taken :)

**Comment by pulsejet** [comment]
> > Hi @pulsejet, precisely. What I propose is to add it to the settings so that the option persists outside the browser. :)
> 
> I second that suggestion :) A reverse chronological order may make sense in a timeline, but not so much in an album, when you want people to explore your photos in the same order as they were taken :)

@ldaverio albums are already sorted chronologically (not reverse), unless you change this in user settings. This issue is about the list of albums, not inside the album itself.

**Comment by ldaverio** [comment]
Oh, sorry, apparently I didn't read the previous posts closely enough, all my apologies.

I'm freshly transitioning from the OwnCloud "Gallery" plugin, and up to now I haven't found how I can share a folder by public link, and have the photos sorted in chronological order. They seem to always come up in reverse order when I open the link as an anonymous user. Also, the "Download all" link seems to be missing, but that's secondary. What matters to me most, is that I can show my photos in the order I took them. 

Again, all my apologies if I misread :(

Laurent.

**Comment by pulsejet** [comment]
@ldaverio folders are not albums.

1. When you share an album, it'll be sorted chronologically.
2. When you share a folder it'll be reverse chronological. If you want it chronologically, add `?sort=album` at the end of the URL.
3. There's no "download all" button for folders anyway. For albums #621

**Comment by eltos** [comment]
> Also, sorting by date is based on the Album's creating date, regardless of when those images were shot.

This is extremely inconvenient if you create albums from older photos. I would even consider this a bug.
Naturally one is interested in the date of a particular event, not the date one found the time to sort the photos into an album.

**Comment by ghimel60** [comment]
> 
> This is extremely inconvenient if you create albums from older photos. I would even consider this a bug. Naturally one is interested in the date of a particular event, not the date one found the time to sort the photos into an album.

I think this is a bug too. This is an important and fundamental feature when I think to an efficient photo management suite...

**Comment by pulsejet** [comment]
@ghimel60 this isn't on the roadmap yet but feel free to open a PR if you can implement it.

**Comment by Phils80** [comment]
Hi, I fully support this idea. I'm aslo missing possibility of sorting list of albums basing on oldest or newest photo of the album. I have lot of them and I'm so desperate that I can use some workaround temporary :) Nowadays album list is sorted over creation date, which is frustrating when you migrate data. 
@pulsejet is there any way to make it manually. Where is stored data creation of the album, can I change it manually somewere ?

**Comment by AlexandruNegura** [comment]
Agree, I was just searching for this functionality since I am moving away from Google Photos. A new entry in that dropdown should be added to sort albums by the date of the last photo. I am manually creating the albums from Photos to keep location metadata and I just figured it out that this sorting is not possible and I have to re-edit each album to sort them correctly. I have ~180 albums so this will be fun :))

**Comment by jcharaoui** [comment]
It would also be great to be able to sort albums chronologically by "most recent photo" in the album.

**Comment by jcharaoui** [comment]
> It would also be great to be able to sort albums chronologically by "most recent photo" in the album.

For the benefit of anyone in the same boat of importing an existing photo collection into Memories and (re-)creating albums with in non-chronological order: here's an SQL query that sets the album creation date date equal to the "date taken" of the most recent photo in each album. Do take a backup of the `nc_photos_albums` before proceeding.

```
UPDATE nc_photos_albums a
INNER JOIN (SELECT af.album_id, UNIX_TIMESTAMP(MAX(m.datetaken)) AS max_datetaken FROM nc_photos_albums_files af INNER JOIN nc_memories m ON af.file_id = m.fileid GROUP BY af.album_id) b
ON a.album_id = b.album_id
SET a.created = b.max_datetaken;
```

**Comment by m-reiter** [comment]
I, too, would love to see that, and have given it a try. After copious amounts of trial and error and with massive help from Github Copilot (another first for me, besides doing anything with PHP or Typescript), I've come up with a version that allows for sorting by oldest or newest photo in an album.

Unfortunately, I also managed to break album creation from within memories. It's too late now to debug that and I could really use some help with that. Any volunteers? @pulsejet Would you have time to take a look?

You can find my code in https://github.com/m-reiter/memories, but I can also open a PR with the unfinished code if that's more convenient.

Thanks in advance for any help!

**Comment by m-reiter** [comment]
Isn't that typical? Just after closing the computer it dawned on me what might be the problem. Two fixes later, it seems to work. Will try to create a PR (my first) for review.

**Comment by RoBo-OnGitHub** [comment]
I'm sharing password protected photo's for years now, and the 'perfect' app for doing so I've still have to discover. Memories is a potential candidate to become it, however when you have a bunch of photo's to tell a story, the sequence is of utmost importance. 

When doing raw filesharing, then order by filename is the most simple; Just add a sequence number as prefix to the existing filenames and you have your custom ordering, while still having access to correct original and modified datetimes. But raw files is not appealing. 
Normally created datetime makes sense, however Memories only does that in newest to oldest. That is not convenient when viewing photos of an event. People going home to people coming in is not how an event is being viewed ;-).

So I think what Memories would need is to have a setting per folder/album to specify the desired ordering in ASC/DESC order.

**Comment by RodSeq** [comment]
> I'm sharing password protected photo's for years now, and the 'perfect' app for doing so I've still have to discover. Memories is a potential candidate to become it, however when you have a bunch of photo's to tell a story, the sequence is of utmost importance.
> 
> When doing raw filesharing, then order by filename is the most simple; Just add a sequence number as prefix to the existing filenames and you have your custom ordering, while still having access to correct original and modified datetimes. But raw files is not appealing. Normally created datetime makes sense, however Memories only does that in newest to oldest. That is not convenient when viewing photos of an event. People going home to people coming in is not how an event is being viewed ;-).
> 
> So I think what Memories would need is to have a setting per folder/album to specify the desired ordering in ASC/DESC order.

Yes I agree, have a setting per folder/album to specify the desired ordering (by Ascending / Descending order, by Creation date / Modified date, by Filename) would actually make Memories the "perfect" flexible app.

Said this, just change the default Memories album/folder/timeline sort order based on files "Creation date" instead of current "Modified date" would always maintain the chronological order of photos & videos as when it was shot (even if I later modify/optimize 1 photo or video in the list post-shooting): would already make Memories the perfect tool for the great majority of video makers / photo makers / event organizers.
